mirage/unikernel.ml

```diff
@@ -490,6 +497,39 @@ module Main (N : Mirage_net.S) (ASSETS : Mirage_kv.RO) = struct
                           in
                           Resolver.update_primary_data resolver trie;
                           Some (`Redirect ("/blocklist", None))))))
+      | `POST (content_type_header, data), "/configuration/upload" -> (
+          let content_type =
+            Option.fold ~none:"multipart/form-data\r\n"
+              ~some:(fun s -> s ^ "\r\n")
+              content_type_header
+          in
+          match Multipart_form.Content_type.of_string content_type with
+          | Error (`Msg e) ->
+              Logs.err (fun m -> m "Bad content-type header: %s" e);
+              Some (`Bad_request ("/configuraton", None))
+          | Ok ct -> (
+              match Multipart_form.of_string_to_list data ct with
+              | Error (`Msg e) ->
+                  Logs.err (fun m -> m "Error parsing form: %s" e);
+                  Some (`Bad_request ("/configuration", None))
+              | Ok (m, assoc) -> (
+                  let multipart_body, _r = to_map ~assoc m in
```

Comment:
isn't this whole multipart stuff now code repetition, and can we move it to a separate function?